tests/test_spend.py

```diff
@@ -364,24 +371,39 @@ def test_coin_selection(lianad, bitcoind):
     assert spend_psbt_2.tx.vin[0].prevout.hash == uint256_from_str(
         bytes.fromhex(spend_txid_1)[::-1]
     )
-    anc_vsize = bitcoind.rpc.getmempoolentry(spend_txid_1)["ancestorsize"]
-    anc_fees = int(
-        bitcoind.rpc.getmempoolentry(spend_txid_1)["fees"]["ancestor"] * COIN
-    )
-    additional_fee = additional_fees(anc_vsize, anc_fees, feerate)
+    additional_fee_at_3satvb = additional_fees(anc_vsize, anc_fees, feerate)
+    assert additional_fee_at_10satvb > additional_fee_at_3satvb
     assert len(spend_res_2["warnings"]) == 1
     assert (
         spend_res_2["warnings"][0]
         == "CPFP: an unconfirmed input was selected. The current transaction fee "
-        f"was increased by {additional_fee} sats to make the average feerate of "
-        "both the input and current transaction equal to the selected feerate."
+        f"was increased by {additional_fee_at_3satvb} sats to make the average "
+        "feerate of both the input and current transaction equal to the selected "
+        "feerate."
     )
 
-    # 2 sat/vb is same feerate as ancestor and we have no warnings:
-    spend_res_2 = lianad.rpc.createspend({dest_addr_2: 10_000}, [], 2)
+    # 2 sat/vb is approx same feerate as ancestor:
+    feerate = 2
+    spend_res_2 = lianad.rpc.createspend({dest_addr_2: 10_000}, [], feerate)
     assert "psbt" in spend_res_2
-    assert len(spend_res_2["warnings"]) == 0
     spend_psbt_2 = PSBT.from_base64(spend_res_2["psbt"])
+    spend_txid_2 = spend_psbt_2.tx.txid().hex()
+    if USE_TAPROOT:
+        assert len(spend_res_2["warnings"]) == 0
+    else:
+        # We still get a warning in the non-taproot case.
```

Comment:
Just checking my understanding of the rationale for updating this: that's because the size of the child is lower under Taproot, and we now correctly pre-estimate its size, so its feerate is slightly larger and therefore it triggers the CPFP logic?